pkgs/native_toolchain_c/lib/src/native_toolchain/recognizer.dart

```diff
@@ -57,30 +51,57 @@ class CompilerRecognizer implements ToolResolver {
   }
 }
 
+Future<Tool?> _whichClang(Uri uri, Logger? logger, Tool? tool) async {
+  final stdout =
+      await CliFilter.executeCli(uri, arguments: ['--version'], logger: logger);
+  if (stdout.contains('Apple clang')) {
+    tool = appleClang;
+  } else {
+    tool = clang;
+  }
+  return tool;
+}
+
 class LinkerRecognizer implements ToolResolver {
   final Uri uri;
+  final OS os;
 
-  LinkerRecognizer(this.uri);
+  LinkerRecognizer(this.uri, this.os);
 
   @override
   Future<List<ToolInstance>> resolve({required Logger? logger}) async {
     final os = OS.current;
     logger?.finer('Trying to recognize $uri.');
     final filePath = uri.toFilePath();
     Tool? tool;
-    if (filePath.contains('-ld')) {
-      tool = gnuLinker;
+
+    //TODO: Make this logic more correct
+    if (filePath.endsWith(os.executableFileName('clang'))) {
```

Comment:
Why do we recognize a linker with a compiler executable name?

Reply:
I'm still confused, why would we use `clang` as a linker? `ld` is the standalone linker. `clang` is a compiler.

What code path tries to use the `LinkerRecognizer` with `clang` instead of with `ld`?